### DISCUSSION TITLE
How to use SwiftGen for internal framework

### DISCUSSION THREAD

**Discussion by tranvutuan** [SwiftGen/SwiftGen#833] [Q&A] [answered]
Currently, our project has multiple targets which is currently sharing some of localization strings. We decide to create a shared framework which is added as dependency to our target. In Cocoapods we are doing 

```
target 'Shared' do
    pod 'SwiftGen'
end
```

After compiling the source codes, `Strings.swift` is generated but I can't access to `L10n` struct since it is marked `internal`. In order to solve this one, I'm planning to  create another public class named `Localization.swift`  in `Shared` framework which will use L10n internally to return strings and these string can be accessible from outside world.

My question is that is it a right way to do it and is it possible to marked L10n as public in this kind of case since it is generated and marked as `internal` by script ?

**Comment by djbe** [comment, marked as answer]
Hey @tranvutuan,

Let me point you to the [strings template documentation](https://github.com/SwiftGen/SwiftGen/blob/stable/Documentation/templates/strings/structured-swift5.md#customization), where there is a section about customisation. Every built-in template we provide has a bunch of (similar) parameters they accept, for handling common scenarios, like yours.

In your case, you'll be interested in the `publicAccess` parameter, you can use it like so:
```yaml
...
strings:
  inputs: ....
  outputs:
    templateName: structured-swift5
    output: Strings.swift
    params:
      publicAccess: true
```

**Reply by tranvutuan** [reply]
@djbe thanks! Let me take a look at it.

**Reply by tranvutuan** [reply]
It works. Thanks. Im trying to mark it as `Answered`but can't do it. Any thoughts ?

**Reply by djbe** [reply]
No clue, I just clicked on the checkmark at the top right of my "answer" box
